Gems/Profiler/Code/Source/CpuProfilerImpl.h

```diff
@@ -64,7 +64,7 @@ namespace Profiler
         // Keeps track of regions that completed (i.e regions that was pushed and popped from the stack)
         // Intermediate storage point for the CachedTimeRegions, when the stack is empty, all entries will be
         // copied to the map.
-        AZStd::fixed_vector<CachedTimeRegion, TimeRegionStackSize> m_cachedTimeRegions;
+        AZStd::vector<CachedTimeRegion> m_cachedTimeRegions;
```

Comment:
To minimize the impact the profiler has on code, it should avoid allocating memory. This is why `fixed_vector` is greatly preferred over `vector`.
There may be three things to do here:
1. `TimeRegionStackSize` can perhaps be increased so `AssetDataStream::Read` fits.
2. `AssetDataStream::Read` cached time regions are reduced to fit inside `TimeRegionStackSize`.
3. The profiler is made more robust against calls that exceed `TimeRegionStackSize`.

Reply:
Another option might be to promote Atom's StableDynamicArray class to AzCore and use that instead. It's gives more control over page size, would have cheaper growth as it doesn't need to move data, and we could even put something in StableDynamicArray::AddPage that would offset the timing data when when the array needs to grow.

Reply:
Would it acceptable to simply ignore all additions to m_CachedTimeRegions once TimeRegionStackSize is reached? This seems more like an edge case and happens only when a level is loaded.

Reply:
Option #1 increases the memory usage by 200 MB.

Reply:
Guarding against exceeding (option 3 and the option @hershey5045 suggest) is probably the best alternative. While using `StableDynamicArray` removes some cost of moving elements (at the cost of more expensive iterations) it still requires allocating memory which for performance is the larger problem.

Reply:
+1 for going with option 3 here as the other options would only be patch fixes

Reply:
I'm running into the profiler running out of timed region memory in AtomSampleViewer outside of a level load. I modified the AtomLoadTest to have 10600 objects, if I try to use the internal profiler to look at the cpu usage it will hit this error in a few seconds.

Reply:
Please take a look at the latest revision. We are now going to discard excess data but I wanted to at least warn users when this happens.